### PR TITLE
use match instead of matchcontinue

### PR DESCRIPTION
**Pull request by hkiel**
modify pushRedeclareIntoExtends3() to make it tail recursive -> saves time and stack

### CODE REVIEW

File: SimulationRuntime/c/simulation/simulation_runtime.cpp [hunk 1]
```diff
@@ -735,6 +735,7 @@ int initRuntimeAndSimulation(int argc, char**argv, DATA *data, threadData_t *thr
   if(!data)
   {
     std::cerr << "Error: Could not initialize the global data structure file" << std::endl;
+    EXIT(0);
```
Comment (sjoelund) [inline SimulationRuntime/c/simulation/simulation_runtime.cpp]: Should be EXIT(1) for failure

**Comment by hkiel** [comment]
How can I start hudson to check this commit?

**Comment by sjoelund** [comment]
You know that you can git commit --amend to avoid adding EXIT(0) and EXIT(1), right? :D (More importantly, I start Hudson manually and changing commits in a PR does not send me an email)

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-02-10_15-08-31](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/1543/).

**Comment by hkiel** [comment]
I found the typo that caused the test suite to fail.
Can you start hudson again, please?

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-02-10_16-44-56](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/1546/console).

**Pull request closed without merging**